File: yh_connect/src/yh_connect_pub.cpp
#include "ros/ros.h"
#include "std_msgs/Int32.h"

int main(int argc, char** argv)
{
    ros::init(argc,argv,"yh_connect_pub");
    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<std_msgs::Int32>("yh_connect_int", 20);

    ros::Rate loop_rate(4);
    std_msgs::Int32 msg;


    while(ros::ok())
    {
        pub.publish(msg);
        msg.data++;
        loop_rate.sleep();
    }
    return 0;
}

File: yh_connect/scripts/yh_connect_pub.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

import rospy
from std_msgs.msg import Int32

class ConnectPub:
    def __init__(self):
        self.pub = rospy.Publisher("yh_connect_int", Int32)
        self.msg = Int32()
        self.loop_rate = rospy.Rate(4)

    def run(self):
        while not rospy.is_shutdown():
            self.pub.publish(self.msg)
            self.msg.data += 1
            self.loop_rate.sleep()

if __name__ == "__main__":
    rospy.init_node("yh_connect_pub")
    connect_pub = ConnectPub()
    connect_pub.run()